src/backend/distributed/planner/distributed_planner.c

```diff
@@ -147,24 +144,10 @@ distributed_planner(Query *parse, int cursorOptions, ParamListInfo boundParams)
 	}
 	else if (CitusHasBeenLoaded())
```

Comment:
Again not related to this PR but while looking at it, wouldn't it be better to throw an error if `CitusHasBeenLoaded` is false at the beginning of this function? Currently we can pass the checks if it is not loaded then it seems to enter an illegal state

Reply:
It is possible to have multiple databases in the same postgres installation, some of which do `CREATE EXTENSION citus;` and some don't. But they share the same `shared_preload_libraries`, which means `distributed_planner` will be called for all of the databases. We don't want to prevent databases which don't want to use Citus from operation.